### PR TITLE
Git basic exercises instructions for reviewer

### LINKED ISSUE #102: Create "##Instructions for reviewer" for project Git Basic Exercises
Please add a section to the bottom of this project's _index.md file.
The section heading should be EXACTLY "## Instructions for reviewer"

Basically we have to set reviewers up for success by telling them what success looks like and what to pay attention to when reviewing this project submission.

For this project, One of the most common problems is merge conflicts. There should be a point that says "If there are merge conflicts then this work is not yet competent" in your instructions.
Description:

- [x]  Look at a bunch of old reviews for this project for each cohort

- [x]  Find the common feedback [This should help you get an idea of what instructions to create]

- [x]  Write them out clearly

- [x]  The section heading should be EXACTLY "## Instructions for reviewer"

### PR DESCRIPTION
**Pull request by Lisa1400**
Related issues: #102 

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by Lisa1400** [comment]
Screenshot from 2021-05-07 11-46-27

File: content/projects/git-exercises/_index.md [hunk 1]
```diff
@@ -403,4 +403,17 @@ Your repo should be all of the following:
...
 
-If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
\ No newline at end of file
+If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
+
+## Instructions For Reviewers
+
+- Ensure that there are 3 branches: main, history, and milkshake-flavours
+- The main branch should consist of 4 files: .gitignore, README.md, history.txt, and milkshakes.md
+- The history branch should consist of 2 files: README.md and history.txt.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: there is an instruction to merge `main` into `history` this comes just before they add the .gitignore, so the history branch should contain the same as the main branch except the gitifnore

File: content/projects/git-exercises/_index.md [hunk 2]
```diff
@@ -403,4 +403,17 @@ Your repo should be all of the following:
 
 We just covered the basics here. Please make sure you understand this stuff. It's super important. Git might seem like a weird theoretical thing to a lot of you. It might seem completely unrelated to the actual job of writing code. But it's not. Git makes teamwork on dev teams possible. Without it we'd spend more time shouting at each other than writing useful code. So learn it. Be comfortable with it. When we start working in teams later on all will be made clear.
 
-If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
\ No newline at end of file
+If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
+
+## Instructions For Reviewers
+
+- Ensure that there are 3 branches: main, history, and milkshake-flavours
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: I would add here that they should look at the network diagram

**Review by ry-oc — COMMENTED** [review]
Left a couple of comments for you :)

**Comment by ry-oc** [comment]
Please also complete the related issues section of the pr, you left it blank implying this PR is not related to any issues

File: content/projects/git-exercises/_index.md [hunk 3]
```diff
@@ -409,9 +409,10 @@ If you are curious now, spend some time googling git branching strategies. We us
...
 - The main branch should consist of 4 files: .gitignore, README.md, history.txt, and milkshakes.md
-- The history branch should consist of 2 files: README.md and history.txt.
+- The history branch should consist of 3 files: README.md, history.txt and milkshakes.md. Consider that main should be merged into history before the .gitignore file is created.
 - The milkshake-flavours branch should consist of 2 files: README.md and milshakes.md.
-- The repo should have a minimum of 9 commits.
+- The repo should have a minimum of 9 commits. 
+- Take a look a the network diagram or graph.
 - On the main branch:
  1. `.gitignore` file should have `ignore-me.db` and `/large-directory-that-should-be-local-only` in it.
  2. `README.md` should have `booya` in it.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: Sorry, one thing I missed, we need to check for merge conflicts on readme

File: content/projects/git-exercises/_index.md [hunk 4]
```diff
@@ -409,9 +409,10 @@ If you are curious now, spend some time googling git branching strategies. We us
 
 - Ensure that there are 3 branches: main, history, and milkshake-flavours
 - The main branch should consist of 4 files: .gitignore, README.md, history.txt, and milkshakes.md
-- The history branch should consist of 2 files: README.md and history.txt.
+- The history branch should consist of 3 files: README.md, history.txt and milkshakes.md. Consider that main should be merged into history before the .gitignore file is created.
 - The milkshake-flavours branch should consist of 2 files: README.md and milshakes.md.
-- The repo should have a minimum of 9 commits.
+- The repo should have a minimum of 9 commits. 
+- Take a look a the network diagram or graph.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: what do we look for on the network diagram?

**Review by ry-oc — CHANGES_REQUESTED** [review]

File: content/projects/git-exercises/_index.md [hunk 5]
```diff
@@ -403,4 +403,18 @@ Your repo should be all of the following:
...
 
-If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
\ No newline at end of file
+If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
+
+## Instructions For Reviewers
+
+- Ensure that there are 3 branches: main, history, and milkshake-flavours
+- The main branch should consist of 4 files: .gitignore, README.md, history.txt, and milkshakes.md
+- The history branch should consist of 3 files: README.md, history.txt and milkshakes.md. Consider that main should be merged into history before the .gitignore file is created.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: Lets see if we can cut this down a bit, you can remove the specifics about what should be in history and milkshakes

File: content/projects/git-exercises/_index.md [hunk 6]
```diff
@@ -403,4 +403,18 @@ Your repo should be all of the following:
...
+- The milkshake-flavours branch should consist of 2 files: README.md and milshakes.md.
+- The repo should have a minimum of 9 commits. Take a look at the network diagram or graph to check the commits that were made.
+- Ensure that there are no merge conflicts on the README.md file.
+- On the main branch:
+ 1. `.gitignore` file should have `ignore-me.db` and `/large-directory-that-should-be-local-only` in it.
+ 2. `README.md` should have `booya` in it. 
+ 3. `history.txt` should have the latest 1000 lines of commands typed on the person's local machine.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: this can be removed as well

File: content/projects/git-exercises/_index.md [hunk 7]
```diff
@@ -403,4 +403,18 @@ Your repo should be all of the following:
...
+ 4. `milkshakes.md` should have 3 milkshake flavours in it.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: and this

File: content/projects/git-exercises/_index.md [hunk 8]
```diff
@@ -403,4 +403,18 @@ Your repo should be all of the following:
 
 We just covered the basics here. Please make sure you understand this stuff. It's super important. Git might seem like a weird theoretical thing to a lot of you. It might seem completely unrelated to the actual job of writing code. But it's not. Git makes teamwork on dev teams possible. Without it we'd spend more time shouting at each other than writing useful code. So learn it. Be comfortable with it. When we start working in teams later on all will be made clear.
 
-If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
\ No newline at end of file
+If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
+
+## Instructions For Reviewers
+
+- Ensure that there are 3 branches: main, history, and milkshake-flavours
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: Add a line to about the network diagram, we generally check that the branches are present and have been merged to main. We also check that main has been merged back into history.

**Review by ry-oc — CHANGES_REQUESTED** [review]
Sorry, more changes based on some feedback I received from Sheena on other PRs

**Review by ry-oc — APPROVED** [review]

**Review by sheenarbw — CHANGES_REQUESTED** [review]
This seems a bit nitpicky. If people are learning git for the fist time then we dont want to foce the to restart the project or break their heads untangling the messes theyare enevetably going to make. We dont need perfection here, we need the students to demonstrate understanding.

What do we need the student to UNDERSTAND here? How can we tell if they understand?

File: content/projects/git-exercises/_index.md [hunk 9]
```diff
@@ -407,9 +407,9 @@ If you are curious now, spend some time googling git branching strategies. We us
...
-- The milkshake-flavours branch should consist of 2 files.
-- The repo should have a minimum of 9 commits. Take a look at the network diagram or graph to check that the branches are present and have been merged into main. Also, check that main has been merged back into history.
-- Ensure that there are no merge conflicts on the README.md file.
+- Check if multiple branches exist, to ensure recruits understand how to create branches.
+- Check if the recruit understands how merging works, by ensuring that the main branch has files merged into it. You can also take a look at the network diagram to check for merging.
+- Check if there are multiple commits.
+- Check the commit messages to see if a recruit actually knows how to add, commit or push using the terminal or command line. If the commit message is `Add files via upload`, it indicates that the command line was not used.
+- Ensure that there are no merge conflict. If this kind of text `<<<<<<< HEAD` or `=======` appears, it indicates that merge conflicts were not resolved.
+- Ensure that a recruit actually had a merge conflict to resolve by using the command `git show`.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: Is this last point necessary?

File: content/projects/git-exercises/_index.md [hunk 10]
```diff
@@ -407,9 +407,9 @@ If you are curious now, spend some time googling git branching strategies. We us
...
-- The history branch should consist of 3 files. Consider that main should be merged into history before the .gitignore file is created.
-- The milkshake-flavours branch should consist of 2 files.
-- The repo should have a minimum of 9 commits. Take a look at the network diagram or graph to check that the branches are present and have been merged into main. Also, check that main has been merged back into history.
-- Ensure that there are no merge conflicts on the README.md file.
+- Check if multiple branches exist, to ensure recruits understand how to create branches.
+- Check if the recruit understands how merging works, by ensuring that the main branch has files merged into it. You can also take a look at the network diagram to check for merging.
+- Check if there are multiple commits.
+- Check the commit messages to see if a recruit actually knows how to add, commit or push using the terminal or command line. If the commit message is `Add files via upload`, it indicates that the command line was not used.
+- Ensure that there are no merge conflict. If this kind of text `<<<<<<< HEAD` or `=======` appears, it indicates that merge conflicts were not resolved.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: Appears where?

File: content/projects/git-exercises/_index.md [hunk 11]
```diff
@@ -407,9 +407,9 @@ If you are curious now, spend some time googling git branching strategies. We us
...
 
-- Ensure that there are 3 branches: main, history, and milkshake-flavours
-- The main branch should consist of 4 files: .gitignore, README.md, history.txt, and milkshakes.md
-- The history branch should consist of 3 files. Consider that main should be merged into history before the .gitignore file is created.
-- The milkshake-flavours branch should consist of 2 files.
-- The repo should have a minimum of 9 commits. Take a look at the network diagram or graph to check that the branches are present and have been merged into main. Also, check that main has been merged back into history.
-- Ensure that there are no merge conflicts on the README.md file.
+- Check if multiple branches exist, to ensure recruits understand how to create branches.
+- Check if the recruit understands how merging works, by ensuring that the main branch has files merged into it. You can also take a look at the network diagram to check for merging.
+- Check if there are multiple commits.
```
Comment (ry-oc) [inline content/projects/git-exercises/_index.md]: Is this point necessary? Is it not common sense for there to be multiple commits?

**Review by ry-oc — CHANGES_REQUESTED** [review]

**Review by ry-oc — APPROVED** [review]

File: content/projects/git-exercises/_index.md [hunk 12]
```diff
@@ -403,4 +403,11 @@ Your repo should be all of the following:
...
 We just covered the basics here. Please make sure you understand this stuff. It's super important. Git might seem like a weird theoretical thing to a lot of you. It might seem completely unrelated to the actual job of writing code. But it's not. Git makes teamwork on dev teams possible. Without it we'd spend more time shouting at each other than writing useful code. So learn it. Be comfortable with it. When we start working in teams later on all will be made clear.
 
-If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
\ No newline at end of file
+If you are curious now, spend some time googling git branching strategies. We use the feature branching strategy here. We'll cover it in depth later on in the course.
+
+## Instructions For Reviewers
+
+- Check if multiple branches exist, to ensure recruits understand how to create branches.
+- Check if the recruit understands how merging works, by ensuring that the main branch has files merged into it. You can also take a look at the network diagram to check for merging.
```
Comment (sheenarbw) [inline content/projects/git-exercises/_index.md]: try not to use the word "recruit". Rather say "learner"

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Other code schools use this content so we need to make use of language that works for other people. 

Don't say recruit, rather say "learner"

**Review by ry-oc — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**